src/tests/functional/plugin/gpu/concurrency/gpu_concurrency_tests.cpp

```diff
@@ -108,3 +108,83 @@ INSTANTIATE_TEST_SUITE_P(smoke_RemoteTensor, OVConcurrencyTest,
     ::testing::Combine(::testing::ValuesIn(num_streams),
         ::testing::ValuesIn(num_requests)),
     OVConcurrencyTest::getTestCaseName);
+
+TEST(OVConcurrencyTest, canSwapTensorsBetweenIRs) {
+    std::vector<std::vector<uint8_t>> ref;
+    std::vector<ov::Tensor> input_tensors;
+    auto fn = ngraph::builder::subgraph::makeSplitMultiConvConcat();
+
+    auto ie = ov::Core();
+    auto compiled_model = ie.compile_model(fn, CommonTestUtils::DEVICE_GPU);
+
+    const int infer_requests_num = 2;
+    ov::InferRequest infer_request1 = compiled_model.create_infer_request();
+    ov::InferRequest infer_request2 = compiled_model.create_infer_request();
+
+    input_tensors.push_back(infer_request1.get_input_tensor());
+    input_tensors.push_back(infer_request2.get_input_tensor());
+
+    auto calc_ref_results = [&](const ov::Tensor& tensor){
+        const auto tensorSize = tensor.get_byte_size();
+        const auto inBlobBuf = static_cast<uint8_t*>(tensor.data());
+        std::vector<uint8_t> inData(inBlobBuf, inBlobBuf + tensorSize);
+        auto reOutData = ngraph::helpers::interpreterFunction(fn, {inData}).front().second;
+        ref.push_back(reOutData);
+    };
+
+    auto compare_results = [&](ov::Tensor& result, const uint8_t* refResult) {
+        auto thr = FuncTestUtils::GetComparisonThreshold(InferenceEngine::Precision::FP32);
+        ASSERT_EQ(ov::shape_size(fn->get_output_shape(0)), result.get_size());
+        FuncTestUtils::compareRawBuffers(result.data<float>(),
+                                        reinterpret_cast<const float *>(refResult), ov::shape_size(fn->get_output_shape(0)),
+                                        ov::shape_size(fn->get_output_shape(0)),
+                                        thr);
+    };
+
+    for (size_t i = 0; i < infer_requests_num; i++) {
+        FuncTestUtils::fill_tensor(input_tensors[i], 10, -5, 1, i);
+        calc_ref_results(input_tensors[i]);
+    }
+
+    // Swap tensors between IRs
+    infer_request1.set_input_tensor(input_tensors[1]);
+    infer_request2.set_input_tensor(input_tensors[0]);
```

Comment:
can we add a test with swapping outputs also?

Reply:
Done